### PR TITLE
Fix #115: cmk ignores the CLI profile arg when loading the cache file

### LINKED ISSUE #115: cmk ignores the CLI profile arg when loading the cache file
### Summary
Cloudmonkey requires a default profile in `.cmk/config` to read the `.cmk/profiles/*.cache` files and it will ignore the command line arg when loading said cache, preferring the default profile.

### System
OS: MacOS 11
CPU: darwin.x86-64
CMK Version: 6.2.0 (build: 8aae61e, 2021-09-23T10:50:26+0530)

### Setup
Create a sample `.cmk/config` with something like:
```
profile = dev
[dev]
...
[prod]
...
```
Then perform a sync:
```
$ cmk sync
Discovered 194 APIs
```
Verify that it did create the sync cache for `dev`:
```
$ ls ~/.cmk/profiles
dev.cache
```
Verify that my custom command did sync and is cached:
```
$ jq '.api[] | select(.name=="myCustomCommand") | {name: .name}' ~/.cmk/profiles/dev.cache   
{
  "name": "myCustomCommand"
}
```

### Attempt 1: Shouldn't work but does
Let's try to use one of those recently synced custom commands for `prod`, which doesn't have any cache file at all (as seen above):
```
cmk -d -p prod myCustomCommand
[debug] Trying to load profile: prod
[debug] cmdline args:cmk, -d, -p, prod, myCustomCommand
[debug] ExecCmd args: myCustomCommand
<actual results of working command>
```
That works, even though it shouldn't. The debug output shows that it is loading `prod` profile but I would expect it to fail since there is no `prod.cache` with the custom commands. Instead, it is reading `dev.cache` for the custom commands and only use `prod` as the profile for the actual execution of the command.

### Attempt 2: Should work, but doesn't
Let's remove the default profile from `.cmk/config` (setting it to `profile =` or entirely deleting the line both do the same):
```
profile =
[dev]
...
[prod]
...
```
Then, since `dev.cache` already exists, let's try and use that:
```
$ cmk -d -p dev myCustomCommand 
[debug] Trying to load profile: dev
[debug] cmdline args:cmk, -d, -p, dev, myCustomCommand
[debug] ExecCmd args: myCustomCommand
Error: unknown command or API requested
```
That _should_ work since there is a `dev.cache` file; there is a `[dev]` profile defined in the `.cmk/config`; and it does recognize that `-p dev` is valid. But because there is no `profile = dev` in `.cmk/config`, it will not work.

### Workarounds
If `myCustomCommand` is the same in all profiles, then I can simply arbitrarily set `profile = dev` in `.cmk/config` and the command line `-p <profile>` will work for all profiles.

Alternatively, I can have separate config files per profile and use the new `-c <config>` option, but that only works starting recently.

### PR DESCRIPTION
**Pull request by nvazquez**
Fixes: #115 

### Tests

````
$ ls ~/.cmk/profiles/
labenv.cache      localcloud.cache
$ grep profile ~/.cmk/config
profile      = 
````

- Load the cache file if the profile configuration is empty

````
$ ./bin/cmk -p labenv list tungstenfabricproviders
$
````

- Load the built-in API cache if there is no cache file for the profile:

````
$ ./bin/cmk -p vmware list tungstenfabricproviders
Loaded in-built API cache. Failed to read API cache, please run 'sync'.
🙈 Error: unknown command or API requested
````

### CODE REVIEW

File: cmk.go [hunk 1]
```diff
@@ -70,6 +82,10 @@ func main() {
 	}
 
 	if *profile != "" {
+		if !existsProfileCache(*profile, cfg) {
+			fmt.Printf("Cannot find a cache file for the profile: %s\n", *profile)
+			os.Exit(1)
```
Comment (rohityadavcloud) [inline cmk.go:87]: I’m not sure if we should exit if cache file isn’t available?

Comment (rohityadavcloud) [reply cmk.go:87]: I would have thought of it using a cache file otherwise giving a warning to the user when not found, and when it’s found just use it.

**Comment by rohityadavcloud** [comment]
What are your thoughts on the expected behaviour - @borisstoyanov @vladimirpetrov @kiranchavala @shwstppr @nvazquez ?

File: cmk.go (hunk 1 above)
Comment (borisstoyanov) [reply cmk.go:87]: agree with the approach

**Comment by borisstoyanov** [comment]
Agree with your suggestion @rohityadavcloud

**Comment by rohityadavcloud** [comment]
Thanks @borisstoyanov @nvazquez - I think the fix can be simply:
- If cache is available for a profile, then load it and use it
- If cache isn't available for a profile, then log a warning message (to stderr) to run the sync command but fallback to the default/inbuilt cache

I was also thinking, what about stale and old caches. For example, I think in the login or API response do we send the ACS version - perhaps another minor idea could be to prompt the user will same message (of running sync) if the API cache is of a different ACS version than the ACS version to which an API call has been made. (just a though)

File: cmk.go (hunk 1 above)
Comment (nvazquez) [reply cmk.go:101]: Thanks, fixing it

**Comment by nvazquez** [comment]
@rohityadavcloud @borisstoyanov can you please review it after the latest changes?

**Review by rohityadavcloud — APPROVED** [review]

**Review by borisstoyanov — APPROVED** [review]
LGTM,

```
➜  cloudstack-cloudmonkey git:(fix-issue-115) rm ~/.cmk/profiles/admin.cache
➜  cloudstack-cloudmonkey git:(fix-issue-115) rm ~/.cmk/profiles/localcloud.cache
➜  cloudstack-cloudmonkey git:(fix-issue-115) ./bin/cmk
Loaded in-built API cache. Failed to read API cache, please run 'sync'.
Apache CloudStack 🐵 CloudMonkey 6.3.0
Report issues: https://github.com/apache/cloudstack-cloudmonkey/issues

(admin) 🐱 > sync
Discovered 598 APIs
(admin) 🐱 > exit
➜  cloudstack-cloudmonkey git:(fix-issue-115) ./bin/cmk
Apache CloudStack 🐵 CloudMonkey 6.3.0
Report issues: https://github.com/apache/cloudstack-cloudmonkey/issues

(admin) 🐱 >
```

**Pull request merged**